pennylane/transforms/qcut/cutcircuit.py

```diff
@@ -365,43 +365,63 @@ def circuit(x):
                 "installed using:\npip install opt_einsum"
             ) from e
 
-    g = tape_to_graph(tape)
-
-    if auto_cutter is True or callable(auto_cutter):
-        cut_strategy = kwargs.pop("cut_strategy", None) or CutStrategy(
-            max_free_wires=len(device_wires)
-        )
-
-        g = find_and_place_cuts(
-            graph=g,
-            cut_method=auto_cutter if callable(auto_cutter) else kahypar_cut,
-            cut_strategy=cut_strategy,
-            **kwargs,
+    tapes, tapes_fn = ([tape], lambda x: x[0])  # pylint: disable=unnecessary-lambda-assignment
+    if isinstance(tape.measurements[0].obs, qml.Hamiltonian):
+        tapes, tapes_fn = qml.transforms.hamiltonian_expand(tape, group=False)
+
+    def res_tapes_fn(results, cut_func=None, indices=None):
+        """Modified post-processing function"""
+        exp_res = [
+            cut_func[idx - 1](results[indices[idx - 1] : indices[idx]])
+            for idx in range(1, len(indices))
+        ]
+        return tapes_fn(exp_res)
+
+    res_tapes, res_funcs, res_index = (), [], [0]
+    for tape in tapes:  # pylint: disable=redefined-argument-from-local
+        g = tape_to_graph(tape)
+
+        if auto_cutter is True or callable(auto_cutter):
+            cut_strategy = kwargs.pop("cut_strategy", None) or CutStrategy(
+                max_free_wires=len(device_wires)
+            )
+
+            g = find_and_place_cuts(
+                graph=g,
+                cut_method=auto_cutter if callable(auto_cutter) else kahypar_cut,
+                cut_strategy=cut_strategy,
+                **kwargs,
+            )
+
+        replace_wire_cut_nodes(g)
+        fragments, communication_graph = fragment_graph(g)
+        fragment_tapes = [graph_to_tape(f) for f in fragments]
+        fragment_tapes = [
+            qml.map_wires(t, dict(zip(t.wires, device_wires))) for t in fragment_tapes
+        ]
+        expanded = [expand_fragment_tape(t) for t in fragment_tapes]
+
+        configurations = []
+        prepare_nodes = []
+        measure_nodes = []
+        for tapes, p, m in expanded:
+            configurations.append(tapes)
+            prepare_nodes.append(p)
+            measure_nodes.append(m)
+
```

Comment:
Can we add a few comments here to understand the additions? The qcut stuff is overall lacking in docs, so anything you've understood as part of this development process will help in the long run :)